src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -499,7 +503,25 @@ void Snippet::SnippetJitExecutor::schedule_6d(const std::vector<MemoryPtr>& inMe
             int64_t indexes[] = {d0, d1, d2, d3, d4};
             jit_snippets_call_args call_args;
             update_ptrs(call_args, inMemPtrs, outMemPtrs);
+#ifdef __linux__
+            if (enable_err_detector) {
+                __sighandler_t signal_handler = [](int signal) {
+                    std::lock_guard<std::mutex> guard(err_print_lock);
+                    if (auto err_emitter = ov::intel_cpu::g_snippets_err_handler->local())
+                        err_emitter->print_debug_info();
+                    auto tid = parallel_get_thread_num();
+                    OPENVINO_THROW("Segfault was caught by the signal handler in snippets node execution on thread " + std::to_string(tid));
+                };
+                struct sigaction new_handler{};
+                new_handler.sa_handler = signal_handler;
+                sigaction(SIGSEGV, &new_handler, nullptr);
```

Comment:
Is it possible to extract sigaction setting from the 5D loop? It seems like handler initialization inside the `parallel_for5d` is overkill. The same for `schedule_nt`

Reply:
I moved to outside of loop, thanks.